src/main/java/org/folio/rest/impl/PurchaseOrderHelper.java

```diff
@@ -211,10 +210,34 @@
     return future;
   }
 
+  private CompositePurchaseOrder populateOrderSummary(CompositePurchaseOrder compPO) {
+    List<CompositePoLine> compositePoLines = compPO.getCompositePoLines();
+    compPO.setTotalEstimatedPrice(calculateTotalEstimatedPrice(compositePoLines));
+    compPO.setTotalItems(calculateTotalItemsQuantity(compositePoLines));
+
+    return compPO;
+  }
+
   private int calculateTotalItemsQuantity(List<CompositePoLine> poLines) {
     return poLines.stream().mapToInt(HelperUtils::calculateTotalQuantity).sum();
   }
 
+  /**
+   * Calculates PO's estimated price by summing the Estimated Price of the associated PO Lines.
+   * See MODORDERS-181 for more details.
+   * @param poLines list of composite PO Lines
+   * @return estimated purchase order's total price
+   */
+  private Double calculateTotalEstimatedPrice(List<CompositePoLine> poLines) {
+    return poLines.stream()
+      .map(CompositePoLine::getCost)
+      .filter(Objects::nonNull)
+      .mapToDouble(cost -> Optional
+        .ofNullable(cost.getPoLineEstimatedPrice())
```

Comment:
when do we anticipate the totalEstimatedPrice to be null? At this point in code shouldn't it either be already calculated ?

Reply:
This is just to cover the case when some third-party added the data directly to storage instead of working via `mod-orders` but probably you are right and this is not required